src/v1_0/rws_client.cpp

```diff
@@ -250,20 +250,43 @@ std::string RWSClient::generateFilePath(const FileResource& resource)
   return Services::FILESERVICE + "/" + resource.directory + "/" + resource.filename;
 }
 
+// Try finding xml tag with class="code" containing retcode.
+std::optional<int> tryParseRetcode(Poco::AutoPtr<Poco::XML::Document> const& xml)
+{
+  std::string code_str = xmlFindTextContent(xml, XMLAttribute{"class", "code"});
+  std::optional<int> result;
+  if (!code_str.empty())
+  {
+    try
+    {
+      result = std::stoi(code_str);
+    }
+    catch (std::invalid_argument) {}
+    catch (std::out_of_range) {}
+  }
+  return result;
+}
 
 POCOResult RWSClient::httpGet(const std::string& uri,
   std::set<Poco::Net::HTTPResponse::HTTPStatus> const& accepted_status)
 {
   POCOResult const result = http_client_.httpGet(uri);
 
   if (accepted_status.find(result.httpStatus()) == accepted_status.end())
-    BOOST_THROW_EXCEPTION(ProtocolError {"HTTP response status not accepted"}
+  {
+    auto exception = ProtocolError {"HTTP response status not accepted"}
       << HttpMethodErrorInfo {"GET"}
       << UriErrorInfo {uri}
       << HttpStatusErrorInfo {result.httpStatus()}
       << HttpResponseContentErrorInfo {result.content()}
-      << HttpReasonErrorInfo {result.reason()}
-    );
+      << HttpReasonErrorInfo {result.reason()};
+    try
+    {
+      if (auto retcode = tryParseRetcode(parseXml(result.content())))
+        exception << RetcodeErrorInfo {retcode.value()};
+    } catch (...) {}
```

Comment:
1. You can avoid code duplication by moving `parseXml()` with `try-catch` to `tryParseRetcode()`
2. Catching everything `(...)` should be avoided when possible, because it hides errors. Catch the exception that you expect from `parseXml()` when the string cannot be parsed.
```suggestion
    if (auto retcode = tryParseRetcode(result.content()))
      exception << RetcodeErrorInfo {retcode.value()};
```